server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -202,6 +204,19 @@ public static ByteSizeValue parseBytesSizeValue(String sValue, ByteSizeValue def
         if (sValue == null) {
             return defaultValue;
         }
+        switch (sValue) {
+            case "0":
+            case "0b":
+            case "0B":
+                return ZERO;
+            case "1b":
+            case "1B":
+                return ONE;
```

Comment:
This is correct but looks like an accident, suggest a comment here too.

```suggestion
                // "1" is deliberately omitted, the units are required for all values except "0" and "-1"
                return ONE;
```